src/main/java/frc/robot/commands/vision/CalculateStdDevs.java

```diff
@@ -0,0 +1,151 @@
+package frc.robot.commands.vision;
+
+import java.util.ArrayList;
+
+import org.photonvision.EstimatedRobotPose;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.constants.Constants;
+import frc.robot.subsystems.Drivetrain;
+import frc.robot.util.LogManager;
+import frc.robot.util.Vision;
+
+/**
+ * Calculates standard deviations for vision
+ */
+public class CalculateStdDevs extends CommandBase {
+  private final ShuffleboardTab m_tab;
+  private final Drivetrain m_drive;
+  private final Vision m_vision;
+  private ArrayList<Pose2d> m_poses;
+  private int m_arrayLength;
+  private Timer m_endTimer;
+
+  /**
+   * Constructor for CalculateStdDevs
+   * @param time How long to run the command, also how many poses to use
+   * @param shuffleboardTab The shuffleboard tab to output to
+   * @param drive The drivetrain
+   * @param vision The vision
+   */
+  public CalculateStdDevs(int time, ShuffleboardTab shuffleboardTab, Drivetrain drive, Vision vision) {
+    m_tab = shuffleboardTab;
+    m_drive = drive;
+    m_vision = vision;
+    m_arrayLength = time;
+    m_endTimer = new Timer();
+  }
+
+  /**
+   * Resets the pose array
+   */
+  @Override
+  public void initialize() {
+    m_poses = new ArrayList<Pose2d>();
+  }
+
+  /**
+   * Adds a pose to the array
+   */
+  @Override
+  public void execute() {
+    Pose2d pose = m_vision.getPose2d(m_drive.getPose());
```

Comment:
just a thought but wouldn't the raw transformation from the camera be a better data point then the posibly inacurate odometry position, epecialy if you don't disable vision from effecting the pose estimator.

Reply:
This uses Vision.getPose2d(), so it doesn't use the drivetrain. I also used the drivetrain as a reference and didn't disable vision to simulate how the robot will use vision on the field.